pyqtgraph/colorRegistry.py

```diff
@@ -0,0 +1,363 @@
+        # print('= hex:', qpen.color().name() )
+
+    def _update_QBrush(self, qbrush, desc):
+        """ updates qbrush to match descriptor for current palette """
+        # print('updating brush',qbrush,'to match',desc)
+        name, alpha = desc
+        if name[0] != '#':
+            if alpha is None:
+                qbrush.setColor( self.color_dic[name] ) # automatically copies
+            else:
+                qcol = QtGui.QColor(self.color_dic[name]) # make a copy
+                qcol.setAlpha(alpha)
+                qbrush.setColor(qcol)
+        else: # set from hex string instead:
+            qcol = QtGui.QColor(name)
+            if alpha is not None: qcol.setAlpha(alpha)
+            qbrush.setColor(qcol)
+        # print('= hex:', qbrush.color().name(), qbrush.color().alpha() )
+        
+
+    def register(self, obj, desc):
+        """
+        Registers obj (QColor, QPen or QBrush) to be updated according to the descriptor on palette change
+        """
+        if hasattr(obj,'registration'):
+            registration = obj.registration
+        else:
+            registration = next(ColorRegistry._registrationGenerator)
+            obj.registration = registration # patch in attribute
+        fin = weakref.finalize(obj, self.unregister, registration)
+        fin.atexit = False # no need to clean up registry on program exit
+        # print('registering', registration, '(',str(obj),'):',str(desc))
+        self.registered_objects[registration] = (weakref.ref(obj), desc)
+
+    def unregister(self, registration):
+        """
+        Removes obj (QColor, QPen or QBrush) from the registry, usually called by finalize on deletion
+        """
+        obj, desc = self.registered_objects[registration]
+        # print('unregistering', registration, '(',str(obj),'):',str(desc))
+        del self.registered_objects[registration]
+        del obj, desc
+
+
+    def colors(self):
+        """ return current list of colors """
+        return self.color_dic # it would be safer (but slower) to provide only a copy
+        
+    def defaultColors(self):
+        """ return a copy of the default color dictionary """
+        return DEFAULT_COLORS.copy()
+    
+
+    def redefinePalette(self, colors=None):
+        """ 
+        Update list of named colors if 'colors' dictionary is given
+        Emits paletteHasChanged signals to color objects and widgets, even if color_dict is None 
+        """
+        if colors is not None:
+            for key in DEFAULT_COLORS:
+                if key not in colors:
+                    raise ValueError("Palette definition is missing '"+str(key)+"'")
+            if DEBUG: print('  ColorRegistry: Setting palette, all color definitions are present.')
+            self.color_dic.clear()
+            self.color_dic.update(colors)
+
+        # notifies named color objects of new assignments:
+        for key in self.registered_objects:
+            ref, desc = self.registered_objects[key]
```

Comment:
can probably just loop over the values directly here?

```python
for ref, desc in self.registered_objects.values():
    ...
```

Reply:
Never thought about that before. Saves a line here and look scleaner. --> done